### ISSUE TITLE
Add `radius filter` or `time filter` around a selected location

### ISSUE THREAD

**Issue by ryanbaumann** [mapbox/mapboxgl-powerbi#96]
### Feature Request

Add a `radius` filter or `drive/walk/cycle` time filter at a selected location on the map.  Highlight and select all user data that falls within the radius or polygon region selected.

See the Mapbox Isochrone plugin for an example of how drive/walk/cycle time polygons work: https://blog.mapbox.com/add-isochrones-to-your-next-application-e9e84a62345f

**Comment by kky1** [comment]
Once a radius or selected area on the map is chosen, it would be incredible to use the export feature in Power Bl of the lats and longs in the selected area

**Comment by ryanbaumann** [comment]
@kky1 to clarify, do you mean that youd like to select & filter points in your viz in the  drive/cycle/walk time polygon region?

**Comment by kky1** [comment]
Yes. I'd like to have the ability to draw/select directly on a MapBox map in Power BI, then export all pinpoints inside the selection.

**Comment by ryanbaumann** [comment]
Can you try thr latest beta version (download in the read me) and see if this works in PowerBI with the latest Mapbox visual?  Cc/ @samgehret

**Comment by kky1** [comment]
Hi Ryan. I've downloaded the zip file. Is there a file location where I should place the java script files? Basically, what do I need to do with the zip file so that Power BI sees this function? Thank you for your assistance.

**Comment by ryanbaumann** [comment]
@kky1 you can try the latest beta at this address https://github.com/mapbox/mapboxgl-powerbi/raw/v1.3/dist/mapboxGLMap.pbiviz

**Comment by AlaskanPreston** [comment]
Hi Ryan,

Can we get a copy of this that works for the March 2018 PBIRS edition?

Thanks,

Preston

**Comment by ryanbaumann** [comment]
cc/ @samgehret

**Comment by AlaskanPreston** [comment]
It works, thanks!!

**Comment by kky1** [comment]
Thank you Ryan and Samgehret - it works and such a great upgrade!! Is there anyway to add a functionality when using either of the drawing tools to measure the distance of the radius, etc. being drawn?

**Comment by mmainzer** [comment]
Hey y'all - 

Tried the beta and it's great. Is there an update on this? Still in beta?

**Comment by mmainzer** [comment]
Hey,

Is there an update on adding the drive-time tools to Power BI custom viz?

I downloaded the link above but it just has the polygon and lasso tools like the regular one. Except the Size parameter doesn't do anything.

Has the original file been moved? Or is this not happening anymore?

**Comment by otravers** [comment]
For the record, this can be done with Mapbox in Tableau because the latter supports displaying arbitrary shapes in visuals, which Power BI doesn't do out of the box:
https://thedataism.wordpress.com/2019/06/13/isochrones-with-tableau-and-mapbox/

We can easily create an M function in Power BI to make dynamic calls to the [isochrone API function](https://docs.mapbox.com/api/navigation/#isochrone), but currently we can't dynamically display the GeoJSON object returned by the API (at best we could load it statically via Mapbox Studio, but that's hardly a substitute).

**Comment by mmainzer** [comment]
Yeah Tableau isn’t an option for me however. I ultimately created a custom application with gl is and turf to dynamically render custom geographies.

That being said, samgehret had shown a demo of the mapbox power bi visual generating isochrone a via the API, on both point-click and search result in his twitter feed. But that was almost 9 months ago and no word on any kind of update. Would be super helpful.

**Comment by kky1** [comment]
I work with Mapbox in Power BI and would love to see this feature. The ARCS GIS map in Power BI offers this; however, the export feature of the data (once a radius of miles or drive time is established and filtered) only allows a small number instead of the entire population selected.

Mapbox in Power BI has much easier drawing tools so is still a better mapping choice if my ultimate goal is to draw on a map and export data back out. Would absolutely love to see this feature in a Mapbox upgrade.

**Comment by mmainzer** [comment]
I agree it should be included in the visual. Currently the only way to generate custom geographies on a map in power bi is with the esri tool and it’s very limited and incredibly confusing to use.

But once my users see they can generate an isochrone or buffer in a custom application, they no longer care much about standard geographies.

**Comment by VBADekker** [comment]
Kind of late to the party and please tell me if I'm off-topic but our marketing department asked for a similar feature; where we can, in a filter in a dashboard, select a location and then that we can see how many clients of that location live within an x-radius. Ideally this radius would then be shown on the map as a circle. Don't know if anybody else is interested in this feature though.

**Comment by kky1** [comment]
You can do this in an ARCS GIS map up to 60 minute drive or 30 mile radius. Unfortunately, ARCS limits you to 250 in an export of it, which isn’t so useful. MapBox maps are much more user friendly with the drawing tools on the map.

**Comment by LS170** [comment]
any news on the driving time feature for Power BI visual?
